RecoMuon/MuonSeedGenerator/src/MuonSeedOrcaPatternRecognition.cc

```diff
@@ -211,42 +230,68 @@ void MuonSeedOrcaPatternRecognition::produce(const edm::Event& event, const edm:
 
     MuonRecHitContainer all = muonMeasurements->recHits(ME4Bwd,event);
     MuonRecHitContainer tmp = filterSegments(muonMeasurements->recHits(ME3Bwd,event), endcapdThetaCut);
+    all.reserve(all.size()+tmp.size());
     copy(tmp.begin(),tmp.end(),back_inserter(all));
 
     tmp = filterSegments(muonMeasurements->recHits(ME2Bwd,event), endcapdThetaCut);
+    all.reserve(all.size()+tmp.size());
     copy(tmp.begin(),tmp.end(),back_inserter(all));
 
     tmp = filterSegments(muonMeasurements->recHits(ME12Bwd,event), endcapdThetaCut);
+    all.reserve(all.size()+tmp.size());
     copy(tmp.begin(),tmp.end(),back_inserter(all));
 
     tmp = filterSegments(muonMeasurements->recHits(ME11Bwd,event), endcapdThetaCut);
+    all.reserve(all.size()+tmp.size());
     copy(tmp.begin(),tmp.end(),back_inserter(all));
 
+    if (me0BackwardLayers.size()){
+      const DetLayer* ME0Bwd = me0BackwardLayers[0];
+      tmp = filterSegments(muonMeasurements->recHits(ME0Bwd,event), endcapdThetaCut);
+      all.reserve(all.size()+tmp.size());
+      copy(tmp.begin(),tmp.end(),back_inserter(all));
+    }
+    if (me0ForwardLayers.size()){
+      const DetLayer* ME0Fwd = me0ForwardLayers[0];
+      tmp = filterSegments(muonMeasurements->recHits(ME0Fwd,event), endcapdThetaCut);
+      all.reserve(all.size()+tmp.size());
+      copy(tmp.begin(),tmp.end(),back_inserter(all));
+    }
+    
     tmp = filterSegments(muonMeasurements->recHits(ME11Fwd,event), endcapdThetaCut);
+    all.reserve(all.size()+tmp.size());
```

Comment:
I don't think there is any reason for these reserves. std::copy will do the right thing.